streams/src/test/java/org/apache/kafka/streams/kstream/internals/graph/StreamsGraphTest.java

```diff
@@ -254,6 +253,7 @@ private Topology getTopologyWithChangingValuesAfterChangingKey(final String opti
 
     }
 
+    @SuppressWarnings("deprecation") // specifically testing the deprecated variant
```

Comment:
This would be a case where I would advocate more strongly to deprecate _this_ method, to avoid accidentally "hiding" the deprecation from callers.

Reply:
Well, but then we need to add more suppression or deprecation upstream. Does not seem worth for testing code

Reply:
This is exactly the point!

Reply:
Updated